core/embed/rust/src/storagedevice/storage_device.rs

```diff
@@ -0,0 +1,690 @@
+    let block = || {
+        VERSION
+            .set(Buffer::try_from(version)?.try_into()?)?
+            .try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_is_initialized() -> Obj {
+    let block = || Ok(INITIALIZED.get().unwrap_or(false).into());
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_is_initialized(is_initialized: Obj) -> Obj {
+    let block = || INITIALIZED.set(is_initialized.try_into()?)?.try_into();
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_rotation() -> Obj {
+    let block = || Ok(ROTATION.get().unwrap_or(0).into());
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_rotation(rotation: Obj) -> Obj {
+    let block = || {
+        let rotation = rotation.try_into()?;
+
+        if ![0, 90, 180, 270].contains(&rotation) {
+            Err(Error::ValueError(cstr!("Not valid rotation")))
+        } else {
+            ROTATION.set(rotation)?.try_into()
+        }
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_label() -> Obj {
+    let block = || LABEL.get().map_or(Ok(Obj::const_none()), |x| x.try_into());
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_label(label: Obj) -> Obj {
+    let block = || {
+        let label: StrBuffer = label.try_into()?;
+        LABEL.set(label.try_into()?)?.try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_device_id() -> Obj {
+    let block = || {
+        if let Some(device_id) = DEVICE_ID.get() {
+            device_id.try_into()
+        } else {
+            let new_device_id = random::get_random_bytes::<12>();
+            let hex_id = helpers::hexlify_bytes(&new_device_id);
+            DEVICE_ID.set(hex_id.clone())?;
+            hex_id.try_into()
+        }
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_device_id(device_id: Obj) -> Obj {
+    let block = || {
+        let device_id: StrBuffer = device_id.try_into()?;
+        DEVICE_ID.set(device_id.try_into()?)?.try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_mnemonic_secret() -> Obj {
+    let block = || {
+        MNEMONIC_SECRET
+            .get()
+            .map_or(Ok(Obj::const_none()), |x| (&x as &[u8]).try_into())
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_mnemonic_secret(
+    n_args: usize,
+    args: *const Obj,
+    kwargs: *mut Map,
+) -> Obj {
+    let block = |_args: &[Obj], kwargs: &Map| {
+        let secret: Buffer = kwargs.get(Qstr::MP_QSTR_secret)?.try_into()?;
+        let backup_type: u8 = kwargs.get(Qstr::MP_QSTR_backup_type)?.try_into()?;
+        let needs_backup: bool = if kwargs.contains_key(Qstr::MP_QSTR_needs_backup) {
+            kwargs.get(Qstr::MP_QSTR_needs_backup)?.try_into()?
```

Comment:
just `kwargs.get(...).map(TryInto::try_into).ok_or(false)`

dtto below

Reply:
Done in [ba44c83](https://github.com/trezor/trezor-firmware/pull/2214/commits/ba44c834a8c0c826cc6480f6e5c1c84a09fc12fa)